pkg/inject/secrets-provider.go

```diff
@@ -6,30 +6,54 @@ import (
 )
 
 type SecretsProviderSidecarConfig struct {
-	containerMode           string
-	containerName           string
-	sidecarImage            string
-	secretsDestination      string
+	containerMode      string
+	containerName      string
+	sidecarImage       string
+	secretsDestination string
 }
-var validEnvVars = []string{
+
+var conjurEnvVars = []string{
 	"CONJUR_ACCOUNT",
+	"conjurAccount",
 	"CONJUR_APPLIANCE_URL",
+	"conjurApplianceUrl",
 	"CONJUR_AUTHENTICATOR_ID",
+	"authnK8sAuthenticatorID",
 	"CONJUR_AUTHN_URL",
 	"CONJUR_SSL_CERTIFICATE",
+	"conjurSslCertificate",
+}
+
+func getConjurAuthnURL(envVars map[string]string) string {
+
+	authnMethod := "authn-k8s"
+	// If "CONJUR_AUTHN_URL" is explicitly set, use it
+	if envVars["CONJUR_AUTHN_URL"] != "" {
+		return envVars["CONJUR_AUTHN_URL"]
+	}
+	return envVars["conjurApplianceUrl"] + "/" + authnMethod + "/" +
+		envVars["authnK8sAuthenticatorID"]
```

Comment:
To fully replace the intermediate ConfigMap, I think we need a way for the user to specify the `authnMethod` without providing the entire `CONJUR_AUTHN_URL`. This would probably require reworking the Golden ConfigMap in some way, so maybe this is sufficient for now.

Reply:
It's not replacing the intermediate ConfigMap, but reading either one. So the user could supply the Golden config or the intermediate config, which ever is more convenient for the user.